src/databricks/labs/ucx/progress/workflow_runs.py

```diff
@@ -0,0 +1,79 @@
+import datetime as dt
+import logging
+from dataclasses import dataclass
+
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.sdk.errors import NotFound
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class WorkflowRun:
+    started_at: dt.datetime
+    """The timestamp of the workflow run start."""
+
+    finished_at: dt.datetime
+    """The timestamp of the workflow run end."""
+
+    workspace_id: int
+    """The workspace id in which the workflow ran."""
+
+    workflow_name: str
+    """The workflow name that ran."""
+
+    workflow_id: int
+    """"The workflow id of the workflow that ran."""
+
+    workflow_run_id: int
+    """The workflow run id."""
+
+    workflow_run_attempt: int
+    """The workflow run attempt."""
+
```

Comment:
As noted [here](https://github.com/databrickslabs/ucx/issues/2600#issuecomment-2383202696), we also need to have the snapshot identifier to disambiguate between crawls that have occurred within a single workflow run:
```suggestion

    snapshot_id: int
    """The unique identifier of the snapshot that took place, used to identify the records associated with the same snapshot."""
```

Reply:
as per https://github.com/databrickslabs/ucx/pull/2743#pullrequestreview-2341125401, we don't need `snapshot_id`

Reply:
We decided to not add the `snapshot_id` to the workflow runs table